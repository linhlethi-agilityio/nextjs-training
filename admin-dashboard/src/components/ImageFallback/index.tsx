'use client';

import Image, { ImageProps } from 'next/image';
import { memo, useState } from 'react';

interface ImageFallbackProps extends ImageProps {
  src: string;
  fallbackSrc: string;
  alt: string;
}

const ImageFallback = ({
  src,
  fallbackSrc,
  alt,
  ...rest
}: ImageFallbackProps) => {
  const [imgSrc, setImgSrc] = useState(src);

  return (
    <Image
      src={imgSrc}
      blurDataURL={fallbackSrc}
      placeholder="blur"
      onLoadingComplete={(result) => {
        if (result.naturalWidth === 0) {
          // Broken image
          setImgSrc(fallbackSrc);
        }
      }}
      onError={() => {
        setImgSrc(fallbackSrc);
      }}
      alt={alt}
      {...rest}
    />
  );
};

export default memo(ImageFallback);
