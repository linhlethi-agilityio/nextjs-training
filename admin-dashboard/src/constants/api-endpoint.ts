export const API_ENDPOINT = {
  ORDERS: 'orders',
};
