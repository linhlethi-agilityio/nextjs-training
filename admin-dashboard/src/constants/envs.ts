export const ENVS = {
  API_URL: process.env.NEXT_PUBLIC_BASE_URL,
};
