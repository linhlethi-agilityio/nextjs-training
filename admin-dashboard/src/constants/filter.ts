export enum SORT_BY {
  ID_ORDER = 'idOrder',
  STATUS = 'status',
  RELEASE_DATE = 'createdAt',
  DEADLINE = 'deadline',
  PRICE = 'price',
}

export enum SORT_ORDER {
  ASC = 'asc',
  DESC = 'desc',
}
