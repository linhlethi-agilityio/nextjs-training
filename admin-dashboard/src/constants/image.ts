export const BLUR_DATA_URL =
  'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gHYSUNDX1BST0ZJTEUAAQEAAAHIAAAAAAQwAABtbnRyUkdCIFhZWiAH4AABAAEAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAACRyWFlaAAABFAAAABRnWFlaAAABKAAAABRiWFlaAAABPAAAABR3dHB0AAABUAAAABRyVFJDAAABZAAAAChnVFJDAAABZAAAAChiVFJDAAABZAAAAChjcHJ0AAABjAAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAAgAAAAcAHMAUgBHAEJYWVogAAAAAAAAb6IAADj1AAADkFhZWiAAAAAAAABimQAAt4UAABjaWFlaIAAAAAAAACSgAAAPhAAAts9YWVogAAAAAAAA9tYAAQAAAADTLXBhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABtbHVjAAAAAAAAAAEAAAAMZW5VUwAAACAAAAAcAEcAbwBvAGcAbABlACAASQBuAGMALgAgADIAMAAxADb/2wBDABQODxIPDRQSEBIXFRQYHjIhHhwcHj0sLiQySUBMS0dARkVQWnNiUFVtVkVGZIhlbXd7gYKBTmCNl4x9lnN+gXz/2wBDARUXFx4aHjshITt8U0ZTfHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHz/wAARCAJ3A7MDASIAAhEBAxEB/8QAGgABAQEBAQEBAAAAAAAAAAAAAAECAwYFBP/EABgQAQEBAQEAAAAAAAAAAAAAAAARARIC/8QAFwEBAQEBAAAAAAAAAAAAAAAAAAECA//EABYRAQEBAAAAAAAAAAAAAAAAAAARAf/aAAwDAQACEQMRAD8A+wAw0AAAgAACKgAAIACAAgAIioAioCAAiKgIioKIqAiKgIioCIqKIioCIqAiKgImqgImqyCIqAiKgJqKgIiooiKgqIqAgAIiooiKgCKgIAAiooIqAAAAKAAAAAAKIoAAigAAAoAKAAqKAAIKigAAKigAAoigAAAAAAAAAAAA9YA5AIAqAAAAgAIqAAgAIACAIqAIAIioAgAiKgqAgCKgIioCIqKIioCIIAgAiKyCIqAiKgIioCIqAiKiiIqAiKgqIqAICiIqAIAIAAgKCKgAAACgAAAAAAqAigAoigAAoACooAAKIogqAKAAqAKAAACiAKAAAAAAAAAD1YDkAAAICoAAIAAAgAIACACAgCKgCACIqAIIKIqAiKgIiooiKgIioCIqAiKgIioCIqAiKyAggCCAIIoIIAggoioCIqKCACAgCKgAIoAAAgACgACiAKAAAIoigKgCiKAqKAqAKIogqAKACiKAACiAKAAAAqAKIAoAAAAAPVAOQAACAKgAAgAACAAgAIACCAAgCACAgCACIqCiCAIIoIIAggCCAIIAggIioCIqAiKgIioCIqKIioCIqCoioAgigioAgAgIoAAIAAAAgoogCgAAAKgIoAKIAoAKACiAKAIoigKgCgAogCiKAAAqAKAAAAAAAAAD1Qg5CiAKIAAgKIAAgAICoICoIACAIACCAAgCCAIqCoCAIIoIIAggCCAIIAggDKoCIqAiKgIiooiKgIioCIqAgIKICiIqAIACAoIACAAAACKKIoAAAAiiKAqAKIoCoAoAKIogqAKACiAKACiAKAAqAKIoAAAAKIAogCiAPVCDkAICiAKIAAgKgACAAgAIAIACCAAgCCAAigggCAKiKgCCAIICAgCCAIIAyqAiKgIioCIqAiKiiIrICKgIioKICiIqAIACCKAICoAAICiCiiAKICKACiAKIoKIoCoAoiiKIoCoAoigogCgAogCgAogCiAigAAAogCiAKIA9SIOSqiAKIAogCogCiAAgAIACAAgAggKggCAAgigggKggCCCiCAIIAggCCAIIAggCCAIIAggCCKCCAIAIioAgiqIAICACCgCAAgKIAAKKgAogIogCiKCiANCAKIoiiKCiAKIoKIAqoAoigogIogCgAogCiAKIoAAAAAAPT0SjkqiUBRAFEQFEAVBAURAUQAEABABBAVBABBQQQBBAVBAEEAQQUQQBFQERUBEVAEEAZVARFQERUAQRREVAEEAQQBBFFQQUQQFQRRUEBUAAQBRBUUQBRAFEAaEAVWVBRARoQBVZUFEAaEAVUBFEAVUAURQFQBRAFEUFEAUQBRAFEAUQB6elQclUQoKM0BRAFEqAogCogCoICiICoICoICogoIICoIAggAgAggCCAIIKqIAIIIIIKIIAggCCAIIAgigggCCAIIAgiioIAggKgiqqCAoiCKIKKIAogCiUBRAGhAFVlRFEAaEAVWVBVZoI0IA0IAqsqCiAKrKgogCiKCiAKIAogCiAKIAogD01KlK5KtKlKC0qUoLSs0oLSpSgtSpSgtKlSgtKlKBSpSgUqUoBUqVRUpUoKiUoAlSgqFQBBAVEQFQQBBAEEAEQFREBWRBVREBUEAQRQQQBBAEEAQQBBFFREBUKlBUqUqhSpSgtSpSgtKzSqLSpSgtKlKC1WaUGhmrQaKzSqjSs0qDRWaoNDKiKrNUFEKDQlAapWVBRAGhCiKrNWgolKCqyA0IAogCiAKIAogCiAPS0qUrk0tKlKC0qVKDVKzSgtKlKC0rNKC0qVKC0qVKDVSpSgtSpSqLUqVKC0qVKC1KVKC1KlKBUpUoLUqVKC1CpQEKlBUSpQVBAEEAQQBBAEEAESiqiVFFZEAQQFRKgKiJQWolKoJSpQKVKlBalSpVFpUqUFpWaVRqpWaUGqVmlUapWaUGqVmlBulZpRGqtYq0GqtYq0RqrWKtBqlZq0Gis1ag1Ss0ojSs0oNVayA1SoUFVmlBoSlBoZq0FEAaGQGhAFEAUQBRAFEoD0lKzSubTVKzSg1Ss0oLSpSgtKzSg1UqVKDVKzSgtKzSgtKlSgtKlSgtKlSgtKlSgtSpSgtSpSgUqVKC1KlKBUpUoLUqVKC1KlKBUpUoLUqVKC1KlSgqJSgIVKBUpWaC1KlKoVCpRSpSpQKlSpQWpUqUFqVKlUWpUqUFqVKm6otSpUorVSs1Ko1UqVAapWaKi0qUoVaVKUStUrNKFaq1ilUbq1zq0RurXOrUG6tYpRG6tYq0GqtYq0GqtYq0GqtYq0RqrWKtBqjNWoNUrNKDVWs0oNUrNWgtWs0oNUrNKDVKlKC1azSg1Ss0oNUrNKDVGaA9HSs0rm21Ss0oNUrNKDVKzSgtKlSg1Ss0oLSs0oNVKlSg1UqUoLSs0oLUqUoLUqUoLUqVKDVSpUoLUqUoLUqVKC1KVKC1KlSgtSpSgVKVKC1KlSgtSpUoLUqVKC1KlKBUpUoFSlZoLUqVKotSpUoq1KlTdBalTdSgbqVN1N1Rd1Kzupugu6lSpVFqVKUKtKzSlSrSohUrVSoFRaVAotEFooBQAWgAUWlQVFq1kBurWKUG6tYpRHSlYq0G6tc6tRK3VrnVoVurWKUK3Ss0oVulZpRa1VrFWhWqVmlQrdKxVoVqlZpQapWaUGqVmrQapWaUGqM0B6KlZpXN0apWaUGqVmlBqlZpQaqVKlBqlZpQaqVKlBqlZpQWlZpQWlZpQWlZpQWlZpQWlZpQWpUqUGqlSpQWlSpQWpUqUFpUqUFqVKlBalSlAqUqUFqVKlBalSpQWpUqUFqVKlBalSpQWpUqVQqVN1N0FqbqbrO6Ku6m6zupugu6m6zupQWpUpVSrUqAlVAEAAAAAAAAAAAAAAAAUQWiiBRRFWgUFqLVrIDVWsVaJGqtYq0I1VrNKJGqtZpRI1Ss1aDVKzQGqtZoI1SsgNVawA3SsgNUrNKDVKzSg1RmgPR0rNK5O7VKzSg1Ss0oNUrNKDVSs0oNUrNKDVSs0oNUrNSg1Ss0oLSs0oLSs0oLSs0oLSs0oLSs0oLUqVKDVSpUoLUqUoLUqVKC1KlKC1KlSgtSpUoLUqVKC1KlSgtSpUoLUqVN0F3U3U3Wd0FqbqbrO6DW6zupus7oNbrO6zupVF3UqVAq1AVkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKKILRSoFFq1lVqNUrJQjdKxVokaq1irQjVKzSiRulZpQjVKzShGqVKUSLVrNKEapWaUI1RmgR6KlZpXN2apWaUGqVmpQbqVmlBqlZpQapWalBqlZpQaqVmlBqlZqUGqVmlBaVmlBaVmlBaVmlBaVmpQaqVKlBqpUqUFpWalBqpUqUFqVKlBqpUqUFqVKlBalSpQWpUqUFqVmpug1Wd1N1ndQa3U3Wd1ndBrdZ3Wd1ndWDW6lRFiVagKgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAqALSoLRatZCjVKyFSNVawFI3Ss0pSNUrNKUjVGaFI9DSs0rDbVKzSg1Ss0oNUrNSg3UrNKDVKzSg1UrNKDVKzUoNUrNKDVSs0oNVKlSg1UqVKDVSpUoNVKlSg1UrNKC0rNSg1UqVKDVSs1KDVSpUoLUqVKC1KlTdQWpU3Wd0Gqm6zus7oN7rO6zus76BrdTfTG+kqwa30zUFSgCoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+7Ss0rLTVKzSg1Ss0oNUrFKDdKxSg1Ss0oNUrFKDVKzUoN1KzSg1SsUoNUrNSg1Ss1KDVKzUoNUrNSg1UrNKDVSs1KDVKzUqDVSs1KDVSs1KDVSs1ndQb3Wd1ndTfSjW6m6xvpN9EGt9Jvpio1ErW+kqCoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+xSsUrLTdKxSg3SsUoN0rFKDVKzSg1SsUoN1KzSg1SsUoNUrNSg3UrNSg3UrNKDVSs1Kg3UrNSg3UrNSg3UrFSg3UrHRUo1UrNSg1UrPTO+gbqVjfSb6WDe+md9MUqxK1vpmoLCrUBUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAfTpXOlYadKlYpQbpWKUG6VilBulc6UG6VilBulYqUG6Vip0DdKx0dA3UrHSdA6VKx0nSDpUrHSdA3Tpz6Ogb6TpjpOiDdSsdJ0sHTpOnOpVg3vpOmaixK10lQUqoAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD9nR05dHTm269J059HQOnR059HSjp0dOfSdA69J059HQOnR059J0Qdek6c+jog6dJ059HRB06Tpz6KQdOk6YqVYOnSdMVKQb6OmKVYla6SohErVSoKVagCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAN0rNKjdapWaUK1Ss0oVqlZpRKtKlKFWlSoFapWRUq0qAVaVAKtQBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhGhG4zCNARmEaSBEhFBIkIoEQUCMjRFSMiwgRBYQSIKgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOvJy6wjLo5cnLrCA5cnLpCBXPk5dIQHLkjrEgOcI6cnIOUI6cnIOcI3yclHMb5OSjBGoRakYhGoQIzBYCRkaiRUiCwEQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH7YRqERtmEahBGYRqEFYhG4QGIRuEBiJG4RBiEbiQGIRuEBzhG4QHOJHSEQc4kdIkBzhy6RIUc4kdIQo5xI6RItGIkbiRaRmI1CCRkWEVIgqCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPpQjUIjTMI1CAzCNQgMQjcIDEI1CAzCNQgMQjcSAzEjcIgxCNQgMQjUIDESNwgMRI3CIMRI3CIrnEjpEgMRI3EhRiJHSJCjnEjpEi0c4kdIm4tGIjcSKkZFgqRABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH1oRqERpmEahAZhGoQGYRqJAZhGoQGYRqEBmJGoQGYRqEQYhGoQGIRqEBmJGoQGYkbiQGYkbiRBiEahAYiRuJEGIkbhEViJG4kBiJG4kWjETcbiRaOe4kdIm4tHODe4zFqRkaiKkQAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB9qEWERpIRYQGYRqEBmEahAZhGokBIkahAZhGokBIkahEGYkahAZhGokBmEaiQGYRqJAZhGokBmEaiRBmJG4kBiEaiRBmJG4kRWIkbiRBiJG4kBmJGoRaOcTcdIzFoxuMx0ibjVHODe4zFSMiipEAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAfdgsIjSQiwgMwjQDMI1EBIRSAkSNAMwigJEjQgzEjQDMIsAZhGkBmEaQGYRpAZhGkQZiRqAMxI0kBmEaiRBmJGokFZiRqJGRlI1EBmJG4kBiJG4kUY3E3G4m41Rz3Eje4m4tGEb3EVIyKisgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAosEFCCChBBQg+8KMtJBQEFAQUBEaAZFARGgGRRBEaQERpARGgGUaAZRoBlGgGUaRBkaQGUaQERpEGYkaRFZiRpEGUaiIMxGhRiJG0UYibjcSKMbjO43uJuNUYRvcZVGRRUiACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKogoQAFgAAAKAAAAAAPQCjm0goCCgIKAgoDIoCCiDIoCI0gIjSAgqAgoDIoDIqAiNIgiKAyjSAiNIgyjSAzqNsoqMtpqDAogyjSKMprSKMpuNaijG4m43rO4owje4zGhEaRWYgAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACqACgAAAoACACgAAAAAAAD0Qo5NoKAgoCCgIigIKAiKIIKAyKAiKAiKAiNICI0gIjSAiKIMioCI0gMioCI0iDKNIKyKiDKNJrIiKgIjSKMo0ijKNJqjGpuNJqjGi6jQiNIrKACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACgoAKACgAqAAAAACgAAAAAAAD0gDi2AAIoCCgIKgCKAgqIIKAiKAgqAgqAgqAgqAgqAiKAiKIMioCIqAiNIgiKgqajSIMo0iDKLqIIjSAyjSKMo0ijOprWoozrLWprQyjWooiKKygAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKACgAqACgAAAoAAAAAqiCgIKAgoD0gDg2AAAAgoCAAIoCAICKAiKAgAIACCoCCoCAAiKAiKiCCoCIqAiNIgiKgIioiojTKCajTKCIoDKNIojLSKIy0mqM6mtM6oyjWprQiKgiCorIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAqKoAKACgAqAAACgAACqAAAAAAAAAAPSgODaCgIKgAACKAgCAigIACCoAioAioAioAioAioAioCAIIioAioCIqIIigIioioioCIqazoiKiCIqKIiooiaqaoiaqa0M6mtazqiAKiIqKgAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoDQAKACoAAAKAAACigAAAAAAAAAAA9KA4NgAAAIAAAAgAAIIAAgAIAAAIACAAiAAgAgCCAAiACICCAAiAiogAiaDIiAgiAogCjJoKMpoKJrOg0IAoiArIAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoDQAKACoAAAKAACgoAAAAAAAAAAAA//Z';
