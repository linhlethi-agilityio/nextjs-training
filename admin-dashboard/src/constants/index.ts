export * from './menu';
export * from './routes';
export * from './messages';
export * from './select';
export * from './tab-label';
export * from './api-endpoint';
export * from './envs';
export * from './filter';
export * from './input';
export * from './validate';
