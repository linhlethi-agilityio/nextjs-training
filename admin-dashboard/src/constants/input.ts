export const REQUIRED_FIELDS = [
  'product',
  'customer',
  'status',
  'deadline',
  'price',
];
