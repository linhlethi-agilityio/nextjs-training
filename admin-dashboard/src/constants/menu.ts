import {
  BarChartIcon,
  CategoryIcon,
  DashboardIcon,
  FileTextFillIcon,
  GearIcon,
  PinMapIcon,
} from '@/icons';

// Routes
import { ROUTES } from './routes';

export const MENU = [
  {
    icon: DashboardIcon,
    label: 'Dashboard',
    path: ROUTES.DASHBOARD,
  },
  {
    icon: FileTextFillIcon,
    label: 'Product',
    path: ROUTES.PRODUCT,
  },
  {
    icon: CategoryIcon,
    label: 'Category',
    path: ROUTES.CATEGORY,
  },
  {
    icon: PinMapIcon,
    label: 'Agen',
    path: ROUTES.AGEN,
  },
  {
    icon: BarChartIcon,
    label: 'Analytic',
    path: ROUTES.ANALYTIC,
  },
  {
    icon: GearIcon,
    label: 'Setting',
    path: ROUTES.SETTING,
  },
];
