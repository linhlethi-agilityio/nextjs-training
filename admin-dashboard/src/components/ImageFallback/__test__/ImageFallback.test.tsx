import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Constants
import { BLUR_DATA_URL } from '@/constants';

// Components
import ImageFallback from '..';

const mockProps = {
  src: '',
  alt: 'image',
  fallbackSrc: BLUR_DATA_URL,
};

describe('ImageFallback component', () => {
  it('should match snapshot for ImageFallback', () => {
    const { container } = render(<ImageFallback {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it('switches to fallbackSrc on error', () => {
    const { getByAltText } = render(<ImageFallback {...mockProps} />);

    const image = getByAltText('image') as HTMLImageElement;
    image.onerror?.(new Event('error'));

    expect(image.src).toContain('http://localhost/');
  });

  it('switches to fallbackSrc when loading completes with broken image', () => {
    const { getByAltText } = render(<ImageFallback {...mockProps} />);

    const image = getByAltText('image') as HTMLImageElement;
    image.onload?.({ naturalWidth: 0 } as any); // Simulate broken image

    expect(image.src).toContain('http://localhost/');
  });
});
