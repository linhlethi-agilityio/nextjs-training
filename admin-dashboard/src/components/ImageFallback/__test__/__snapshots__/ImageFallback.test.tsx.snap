// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFallback component should match snapshot for ImageFallback 1`] = `
<div>
  <img
    alt="image"
    data-nimg="1"
    decoding="async"
    src="http://localhost/"
    style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
  />
</div>
`;
